pallets/cash/src/types.rs

```diff
@@ -454,6 +458,28 @@ impl CashPrincipalAmount {
     }
 }
 
+impl TryInto<CashPrincipal> for CashPrincipalAmount {
+    type Error = Reason;
+
+    fn try_into(self) -> Result<CashPrincipal, Reason> {
+        match i128::try_from(self.0) {
+            Ok(v) => Ok(CashPrincipal(v)),
+            Err(_) => Err(Reason::MathError(MathError::SignMismatch)),
+        }
+    }
+}
+
+impl TryInto<CashPrincipalAmount> for CashPrincipal {
```

Comment:
This one's a bit weirder to add since you can do `principal.withdrawable_amount()` or `principal.repayable_amount()` to get the sign you are expecting